### PR TITLE
Add generics notation to MenuInterface

### PR DESCRIPTION
**Pull request by emodric**
Solves errors by PHPStan similar to:

image

### CODE REVIEW

**Comment by emodric** [comment]
Well, this specifically is not about running PHPStan on this repo codebase, but when using `Knp\Menu\MenuInterface` in other packages, and runing PHPStan on those.

**Comment by emodric** [comment]
I forgot to ask, in the meantime, can this be merged any maybe tagged (not super important), so it can be useful for other libs?

**Comment by core23** [comment]
Looks like you have to rebase this PR. 

Hope this can be merged soon, as it would help many projects that are using phpstan / psalm.

**Comment by garak** [comment]
> Looks like you have to rebase this PR.

No need to rebase, for now

**Comment by core23** [comment]
Travis is failing because of some generic phpunit error, that does not happen on the latest 2.x commit: https://travis-ci.org/github/KnpLabs/KnpMenu/builds/579731224

**Comment by emodric** [comment]
@garak Hi! What is the status of this PR? Is this something that is acceptable?

**Review by garak — APPROVED** [review]

File: src/Knp/Menu/ItemInterface.php [hunk 1]
```diff
@@ -8,6 +8,8 @@
  * It roughly represents a single <li> tag and is what you should interact with
  * most of the time by default.
  * Originally taken from ioMenuPlugin (http://github.com/weaverryan/ioMenuPlugin)
+ *
+ * @extends \IteratorAggregate<self>
```
Comment (derrabus) [inline src/Knp/Menu/ItemInterface.php:12]: Shouldn't this be `IteratorAggregate<TKey, TValue>`?

```suggestion
 * @extends \IteratorAggregate<int, self>
```

https://github.com/phpstan/phpstan-src/blob/fe4c770254b49d64816e4b62c2b6d109f88e0832/stubs/iterable.stub#L11-L25

Comment (stof) [reply src/Knp/Menu/ItemInterface.php:12]: actually, keys are strings as they are the names of children.

Comment (derrabus) [reply src/Knp/Menu/ItemInterface.php:12]: Right, I've updated my suggestion. Thanks.

Comment (emodric) [reply src/Knp/Menu/ItemInterface.php]: Thanks @stof @derrabus ! I've commited the suggestion!

**Review by derrabus — APPROVED** [review]

**Review by wouterj — APPROVED** [review]

**Comment by emodric** [comment]
Thanks for the merge @garak ! Can we have a `2.6.1` patch release with the fix? :)

**Comment by garak** [comment]
Here you are: https://github.com/KnpLabs/KnpMenu/releases/tag/2.6.1

**Comment by emodric** [comment]
Thanks @garak :heart:

**Pull request merged**